src/coreclr/nativeaot/Runtime/i386/ExceptionHandling.asm

```diff
@@ -313,9 +321,7 @@ FASTCALL_FUNC  RhpCallCatchFunclet, 0
         ;; ECX still contains the exception object
         ;; EDX: funclet IP
         ;; EAX: funclet EBP
-        call        RhpCallFunclet
-
-ALTERNATE_ENTRY RhpCallCatchFunclet2
+        CALL_FUNCLET Catch
```

Comment:
Does pushing the ebp that the macro does in the middle of a function work well with the debugger stack unwinding  from the actual funclet / the code inside of this macro after the ebp is pushed?

Reply:
That's good question and I will try.

That said, the hack with custom object file that ProjectN used doesn't really transfer well and requires custom code all over the place. Since the thread hijacking works differently from ProjectN I don't expect this to hit the same issues. The NativeAOT unwinder used for exception handling, GC refs, and hijacking, can work with the ebp on the stack.

Reply:
I suppose I can reserve one more stack slot through `FUNCLET_CALL_PROLOGUE` and use that instead of `push / pop`. Would that make sense?

Reply:
Maybe it is ok as is, so if you verify that stepping through that code keeps debugger showing correct stack trace, we can keep it as is.

Reply:
I think I know how to fix it. I'll look at it once I flush the current patch queue.